lib/cucumber/formatter/junit.rb

```diff
@@ -103,7 +103,7 @@ def end_feature(feature_data)
       def create_output_string(test_case, scenario, result, row_name)
         output = "#{test_case.keyword}: #{scenario}\n\n"
         return output if result.ok?(@config.strict)
-        if test_case.keyword == 'Scenario'
+        if test_case.keyword == 'Scenario' && @failing_step_source
```

Comment:
Entering the else clause will print "Example row" in the report, so that is wrong, we need to enter the then clause - but not use `@failing_step_source` unless it exists.